fvm/Grid/AnalyticBRadialGridGenerator.cpp

```diff
@@ -358,36 +358,342 @@ void AnalyticBRadialGridGenerator::InterpolateInputProfileToGrid(
 }
 
 /**
- * Calculate minor radius coordinate 'r' corresponding to the given
- * Cartesian coordinates (x,y,z).
+ * Interpolates the input chaping profile at a single radius r, and extrapolates using 
+ * the maximum and minimum r for which the profile is defined. This is useful for the 
+ * GetRThetaPhiFromCartesian() and GetGradRCartesian() functions used for e.g. the SPI
+ * module in DREAM, as these functions must be able to handle coordinates outside the plasma
  *
- * (The Cartesian coordinate system is oriented such that x and y span
- * the poloidal plane. The origin of x and y is the magnetic axis.)
  */
-void AnalyticBRadialGridGenerator::GetRThetaFromCartesian(real_t* /*r*/, real_t* /*theta*/,
-    real_t /*x*/, real_t /*y*/, real_t /*z*/, real_t /*lengthScale*/, real_t /*startingGuessR*/
+real_t AnalyticBRadialGridGenerator::InterpolateInputProfileSingleExtrap(real_t r,
+    const len_t nProvided, const real_t *xProvided, const real_t *xProvided_r,
+    gsl_spline *spline_x, gsl_interp_accel *spline_acc
 ) {
-	throw FVMException("AnalyticBRadialGridGenerator: This module is currently incompatible with the SPI module.");
+    if(nProvided==1){
+        return xProvided[0];
+    } else {
+        if(r>xProvided_r[nProvided-1])
+            return gsl_spline_eval(spline_x, xProvided_r[nProvided-1], spline_acc);
```

Comment:
I think the GSL spline is defined so that if evaluated in a point contained in ``xProvided_r``, it will return exactly the corresponding value in ``xProvided`` (i.e. the spline passes through all given points exactly).